packages/node/src/behaviors/scenes-management/ScenesManagementServer.ts

```diff
@@ -4,11 +4,1131 @@
+        const entry: ScenesManagement.SceneInfo =
+            infoEntryIndex !== -1
+                ? this.state.fabricSceneInfo[infoEntryIndex]
+                : {
+                      sceneCount: 0, // Will be updated before it is set
+                      currentScene: UNDEFINED_SCENE_ID,
+                      currentGroup: UNDEFINED_GROUP,
+                      sceneValid: false,
+                      remainingCapacity: 0, // Will be updated before it is set
+                      fabricIndex,
+                  };
+        const { sceneCount, remainingCapacity } = this.#countsForFabric(fabricIndex);
+        entry.sceneCount = sceneCount;
+        entry.remainingCapacity = remainingCapacity;
+        if (infoEntryIndex === -1) {
+            this.state.fabricSceneInfo.push(entry);
+        } else {
+            this.state.fabricSceneInfo[infoEntryIndex] = entry;
+        }
+    }
+
+    #countsForFabric(fabricIndex: FabricIndex) {
+        const sceneCount = this.#scenesForFabric(fabricIndex).length;
+        return {
+            sceneCount,
+            remainingCapacity: Math.max(this.#fabricSceneCapacity - sceneCount, 0),
+        };
+    }
+
+    /** Activates the given scene in the fabric scene info, invalidating all others. */
+    #activateSceneInFabricSceneInfo(fabricIndex: FabricIndex, groupId: GroupId, sceneId: number) {
+        for (const infoEntry of this.state.fabricSceneInfo) {
+            if (infoEntry.fabricIndex === fabricIndex) {
+                infoEntry.currentGroup = groupId;
+                infoEntry.currentScene = sceneId;
+                infoEntry.sceneValid = true;
+
+                this.internal.monitorSceneAttributesForFabric = fabricIndex;
+            } else if (infoEntry.sceneValid) {
+                infoEntry.sceneValid = false;
+            }
+        }
+    }
+
+    /** Removes all scenes for a given fabric when the fabric is deleted */
+    #handleDeleteFabric({ fabricIndex }: Fabric) {
+        this.state.sceneTable = deepCopy(this.state.sceneTable).filter(s => s.fabricIndex !== fabricIndex);
+        this.state.fabricSceneInfo = deepCopy(this.state.fabricSceneInfo).filter(s => s.fabricIndex !== fabricIndex);
+        if (this.internal.monitorSceneAttributesForFabric === fabricIndex) {
+            this.internal.monitorSceneAttributesForFabric = null;
+        }
+    }
+}
+
+export namespace ScenesManagementServer {
+    /** Scene Attribute Data format used internally to store scene attribute values */
+    export type SceneAttributeData = { [key: string]: { [key: string]: boolean | number | bigint | null } };
+
+    /** Scene Table Entry including scene values and fabric index */
+    export interface ScenesTableEntry extends Omit<ScenesManagement.LogicalSceneTable, "extensionFields"> {
+        sceneValues: SceneAttributeData;
+        fabricIndex: FabricIndex;
+    }
+
+    /** Scene Table Entry as decorated class for persistence */
+    export class ScenesTableEntry {
```

Comment:
One thing I've considered is just defining fields with "!" suffix and making a private constructor.  I *think* the decorations would still work though not 100% sure.  That would make implementation smaller and ensure class can only be used as an interface.

Anyway I think if you move the fields above here with a "!" the automatic decoration logic won't see them and you can get rid of the unsafe declaration merge complaint.